### ISSUE TITLE
Puppet RVM looks for presence of gpg2, but RVM installer is agostic

### ISSUE THREAD

**Issue by rwilcox** [maestrodev/puppet-rvm#82]
The RVM install script will check if you have *either* `gpg2` *or* `gpg` installed, and use either to validate the GPG signature.

However, currently Puppet RVM **only** checks for gpg2 (and if it's found install the public key). This could lead to a confusing situation / failed provision if you have gpg installed: the installation will fail because puppet RVM hasn't downloaded the key (no gpg2 in the path!) but RVM's installed script wants to check the validity anyway (because it sees gpg in the path).

While this can be worked around (require the rvm GPG resource before Puppet installs RVM), it was.. rather surprising to see it fail and resulted in a lot of head scratching on my front.

TL; DR: can we check for `gpg` in addition to `gpg2` ? And if either is there use the appropriate one to download the public key?

**Comment by domcleal** [comment]
Indeed, on Debian 7 (wheezy, stable), the binary is called "gpg" so this causes failures.

**Comment by TJM** [comment]
For what its worth, gpg generally should be a symlink to gpg2, for example:

* On CentOS 6.x:
```
# ls -l /usr/bin/gpg
lrwxrwxrwx. 1 root root 4 Oct 29 21:00 /usr/bin/gpg -> gpg2
```

* On CentOS 5.x:
```
# ls -l /usr/bin/gpg
-rwxr-xr-x 1 root root 924152 Jan  8  2014 /usr/bin/gpg
```

**Comment by domcleal** [comment]
For completeness, ditto for CentOS 7.0:

    # ls -l /bin/gpg
    lrwxrwxrwx. 1 root root 4 Aug 27 00:07 /bin/gpg -> gpg2

**Comment by TJM** [comment]
Thanks Dominic, As a result of that, maybe we can just check for `gpg` *instead* of `gpg2` as opposed to "in addition to" since that is going to start to get ugly, logic wise.

**Comment by carlossg** [comment]
If only we had tests to run in actual servers,... beaker to the rescue

**Comment by domcleal** [comment]
Thanks @carlossg, seems to work well in production!
